src/box/lua/upgrade.lua

```diff
@@ -1394,9 +1394,25 @@ local function add_instance_names()
     box.space._space:update({box.schema.CLUSTER_ID}, {{'=', 7, format}})
 end
 
+local function add_trigger_to_func()
+    log.info('add trigger to _func')
+    local _space = box.space[box.schema.SPACE_ID]
+    local _func = box.space[box.schema.FUNC_ID]
+    local _vfunc = box.space[box.schema.VFUNC_ID]
+    for _, v in _func:pairs() do
+        if #v == 19 then
+            _func:update(v[1], {{'=', 20, {}}})
+        end
+    end
+    local ops = {{'=', '[7][20]', {name = 'trigger', type = 'array'}}}
+    _space:update({_func.id}, ops)
+    _space:update({_vfunc.id}, ops)
+end
+
 local function upgrade_to_3_0_0()
     store_replicaset_uuid_in_new_way()
     add_instance_names()
+    add_trigger_to_func()
```

Comment:
3.0.0 has already been released. You can't update the past. You must create an upgrade handler for 3.1.0.

Reply:
Sorry, I forgot about the new release. I created new handlers for upgrade and downgrade.